src/loader.js

```diff
@@ -5,155 +5,198 @@ let Promise = require('promise');
 let _ = require('lodash');
 let TopoSort = require('topo-sort');
 
-// We use this closure to create a .create() method which
-// returns a graphViz file for a given component directory
-function generateGraph(dir, vComp) {
-  let tsort = new TopoSort();
-
-  for (let x of _.keys(dir)) {
-    tsort.add(x, dir[x].requires || []);
+/** Check if a value is a flat value or a component definition */
+let isComponent = (def) => {
+  // Check that it's an object
+  if (typeof(def) !== 'object' && def !== null && def !== undefined) {
+    return false;
   }
-
-  for (let x of _.keys(vComp)) {
-    tsort.add(x, vComp[x].requires || []);
+  // Check that is object has a setup function
+  if (!(def.setup instanceof Function)) {
+    return false;
+  }
```

Comment:
overkill to throw error if we have a property that's called setup but not a function?